polyply/src/simple_seq_parsers.py

```diff
@@ -0,0 +1,245 @@
+            elif token in ONE_LETTER_RNA and RNA:
+                resname = ONE_LETTER_RNA[token]
+            else:
+                msg = f"Cannot find one letter residue match for {token}"
+                raise IOError(msg)
+
+            monomers.append(resname)
+
+    # make sure to set the defaults for the DNA and RNA terminals
+    monomers[0] = monomers[0] + "5"
+    monomers[-1] = monomers[-1] + "3"
+
+    seq_graph =  _monomers_to_linear_nx_graph(monomers)
+    return seq_graph
+
+def _identify_nucleotypes(comments):
+    """
+    From a comment found in the ig or fasta file, identify if
+    the sequence is RNA or DNA sequence by checking if these
+    keywords are in the comment lines. Raise an error if
+    none or conflicting information are found.
+
+    Parameters
+    ----------
+    comments: abc.itertable
+        list of comment lines
+
+    Returns
+    -------
+    bool, bool
+        is it DNA, RNA
+
+    Raises
+    ------
+    FileFormatError
+        neither RNA nor DNA keywords are found
+        both RNA and DNA are found
+    """
+    RNA = False
+    DNA = False
+    for comment in comments:
+        if "DNA" in comment:
+            DNA = True
+
+        if "RNA" in comment:
+            RNA = True
+
+    if RNA and DNA:
+        raise FileFormatError("Found both RNA and DNA keyword in comment. Choose one.")
+
+    if not RNA and not DNA:
+        raise FileFormatError("Cannot identify if sequence is RNA or DNA from comment.")
+
+    return DNA, RNA
+
+def parse_ig(filehandle):
+    """
+    Read ig sequence in DNA/AA formatted file. See following link
+    for format:
+
+    https://www.animalgenome.org/bioinfo/resources/manuals/seqformats#IG
+
+    Parameters
+    ----------
+    filehandle: :class:`pathlib.Path`
+    """
+    with open(filehandle) as file_:
+        lines = file_.readlines()
+
+    clean_lines = []
+    comments = []
+    idx = 0
+    for idx, line in enumerate(lines):
+        clean_line, comment = split_comments(line)
+        comments.append(comment)
+        if clean_line:
+            if clean_line[-1] == '1' or clean_line[-1] == '2':
+                ter_char = clean_line[-1]
+                clean_line = clean_line[:-1]
+                clean_lines.append(clean_line)
+                break
+            else:
+                clean_lines.append(clean_line)
+    else:
+        msg = "The sequence is not complete, it does not end with 1 or 2."
+        raise FileFormatError(msg)
+
+    DNA, RNA = _identify_nucleotypes(comments)
+    seq_graph = _parse_plain(clean_lines[1:], DNA=DNA, RNA=RNA)
+
+    if ter_char == '2':
+        nnodes = len(seq_graph.nodes)
+        seq_graph.add_edge(0, nnodes-1)
+        seq_graph.edges[(0, nnodes-1)]["linktype"] = "circle"
+        seq_graph.nodes[0]["resname"] = seq_graph.nodes[0]["resname"][:-1]
+        seq_graph.nodes[nnodes-1]["resname"] = seq_graph.nodes[nnodes-1]["resname"][:-1]
+
+    if idx < len(lines) - 1 and not lines[idx+1].startswith(";"):
```

Comment:
If you have more than 1 sequence, the next line after the first sequence may very much be a comment to tell what that other is. Instead of pilling up the checks, I would rephrase the error message to be more speculative. "There may be more than one sequence in the file. We will only use the first one."